src/ads/ads-controller.js

```diff
@@ -285,11 +295,12 @@ class AdsController extends FakeEventTarget implements IAdsController {
 
   _onAdLoaded(event: FakeEvent): void {
     this._adIsLoading = false;
-    this._ad = event.payload.ad;
+    this._pendingAd = event.payload.ad;
   }
 
   _onAdStarted(event: FakeEvent): void {
     this._ad = event.payload.ad;
+    this._pendingAd = null;
```

Comment:
You need to check if the pending ad is started(could be more than 1 pending ad), it could happen when multiple ad providers exist like bumper and ima.

Reply:
That's true that u can have multiple providers, but at a certain point of time just one ad can be loaded. I don't think we need to complicate it too much.

Reply:
This issue exists, for example, bumper with ima both postroll which one will be in pendingAd?
if bumper is loading after ima postroll, ima won't be exist in pendingAd and then the logic for smart won't work for postroll.